core/record/recordExprParser.js

```diff
@@ -78,18 +79,27 @@ const bindNode = (survey, record, node) => {
   }
 }
 
-const evalApplicableExpression = async (survey, record, nodeCtx, expressions) =>
-  R.head(await evalApplicableExpressions(survey, record, nodeCtx, expressions, true))
+const evalApplicableExpression = async (survey, record, nodeCtx, expressions, convertToNodeValue = false) =>
+  R.head(await evalApplicableExpressions(survey, record, nodeCtx, expressions, true, convertToNodeValue))
 
-const evalApplicableExpressions = async (survey, record, node, expressions, stopAtFirstFound = false) => {
+const evalApplicableExpressions = async (survey, record, node, expressions, stopAtFirstFound = false, convertToNodeValues = false) => {
   const applicableExpressions = await _getApplicableExpressions(survey, record, node, expressions, stopAtFirstFound)
 
   return await Promise.all(
-    applicableExpressions.map(async expression => ({
+    applicableExpressions.map(async expression => {
+      const valueEval = await evalNodeQuery(survey, record, node, NodeDefExpression.getExpression(expression))
+      const value =
+        convertToNodeValues
+          ? R.isNil(valueEval) || R.isEmpty(valueEval)
```

Comment:
These nested ternary operators feels quite confusing. I would hope that there's a clearer way to express the same logic.
At least I've never seen anyone use them like this before. Maybe sometimes It's nested like `a ? b : c ? d : e` but never ` a ? b ? c : d : e` (though even the latter is not ideal in my books, at least not without parentheses)

Reply:
I wonder if `convertToNodeValues` could be made non-conditional, i.e. to always convert or not to convert? I'm not fully clear on the logic yet, but if it were always done in one way then the overall complexity would decrease a little bit.

Reply:
I'll try to make it more readable. It's used in different places and the conversion is not always needed, so we should continue to specify  `convertToNodeValues`.

Reply:
But is the change idempotent? It would simplify things in general to omit conditionals in these kinds of cases if there's no functional difference.

It might be a bit of extra work to do, but if there's no difference in the perceptible speed, I think it would be a good tradeoff.

Reply:
ok, I'll move the conversion where it's needed and remove the condition check